servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -202,6 +223,42 @@ public void onNext(final ServiceDiscovererEvent<ResolvedAddress> event) {
                 }
             }
 
+            @SuppressWarnings("unchecked")
+            private Host<ResolvedAddress, C> createHost(ResolvedAddress addr) {
+                Host<ResolvedAddress, C> host = new Host<>(addr);
+                host.onClose().afterFinally(() ->
```

Comment:
Is that indeed so? `ListenableAsyncCloseable#onClose` javadoc:

```
* Returns a {@link Completable} that is notified once the {@link ListenableAsyncCloseable} was closed.
     * @return the {@code Completable} that is notified on close.
```

My intention was the following:

When the last connection closes, the host is closed (line 513).
Following that, upon closing of the host, the host is removed from the array of active hosts. It needs to be added here because of the visibility of the `activeHostsUpdater` and the cohesion – the LB creates the `Host`s and allows interactions with its own internal state by adding this hook.

Please let me know if that clarifies it. The tests confirm the behaviour is as expected.